src/app/delegation_backend/src/itn_uptime_analyzer/uptime.go

```diff
@@ -0,0 +1,346 @@
+					Bucket: ctx.BucketName,
+					Key:    obj.Key,
+				})
+
+				if err != nil {
+					log.Fatalf("Error getting object from bucket: %v\n", err)
+				}
+
+				defer objHandle.Body.Close()
+
+				objContents, err := io.ReadAll(objHandle.Body)
+				if err != nil {
+					log.Fatalf("Error getting creating reader for json: %v\n", err)
+				}
+
+				err = json.Unmarshal(objContents, &submissionDataYesterday)
+				if err != nil {
+					log.Fatalf("Error unmarshaling bucket content: %v\n", err)
+				}
+
+				if submissionDataYesterday.GraphqlControlPort != 0 {
+					if (identity["public-key"] == submissionDataYesterday.Submitter.String()) && (identity["public-ip"] == submissionDataYesterday.RemoteAddr) && (identity["graphql-port"] == strconv.Itoa(submissionDataYesterday.GraphqlControlPort)) {
+						if lastSubmissionTimeString != "" {
+							lastSubmissionTime, err = time.Parse(time.RFC3339, lastSubmissionTimeString)
+							if err != nil {
+								log.Fatalf("Error parsing time: %v\n", err)
+							}
+						}
+
+						currentSubmissionTime, err := time.Parse(time.RFC3339, submissionDataYesterday.CreatedAt)
+						if err != nil {
+							log.Fatalf("Error parsing time: %v\n", err)
+						}
+
+						if (lastSubmissionTimeString != "") && (currentSubmissionTime.After(lastSubmissionTime.Add(time.Duration(syncPeriod-5) * time.Minute))) && (currentSubmissionTime.Before(lastSubmissionTime.Add(time.Duration(syncPeriod+5) * time.Minute))) {
+							uptimeYesterday = append(uptimeYesterday, true)
+							lastSubmissionTimeString = submissionDataYesterday.CreatedAt
+						} else if lastSubmissionTimeString == "" {
+							uptimeYesterday = append(uptimeYesterday, true)
+							lastSubmissionTimeString = submissionDataYesterday.CreatedAt
+						}
+					}
+				} else {
+					if (identity["public-key"] == submissionDataYesterday.Submitter.String()) && (identity["public-ip"] == submissionDataYesterday.RemoteAddr) {
+						if lastSubmissionTimeString != "" {
+							lastSubmissionTime, err = time.Parse(time.RFC3339, lastSubmissionTimeString)
+							if err != nil {
+								log.Fatalf("Error parsing time: %v\n", err)
+							}
+						}
+
+						currentSubmissionTime, err := time.Parse(time.RFC3339, submissionDataYesterday.CreatedAt)
+						if err != nil {
+							log.Fatalf("Error parsing time: %v\n", err)
+						}
+
+						if (lastSubmissionTimeString != "") && (currentSubmissionTime.After(lastSubmissionTime.Add(time.Duration(syncPeriod-5) * time.Minute))) && (currentSubmissionTime.Before(lastSubmissionTime.Add(time.Duration(syncPeriod+5) * time.Minute))) {
+							uptimeYesterday = append(uptimeYesterday, true)
+							lastSubmissionTimeString = submissionDataYesterday.CreatedAt
+						} else if lastSubmissionTimeString == "" {
+							uptimeYesterday = append(uptimeYesterday, true)
+							lastSubmissionTimeString = submissionDataYesterday.CreatedAt
+						}
+					}
+				}
+			}
+		}
+
+		for paginatorToday.HasMorePages() {
```

Comment:
Why this one inside the loop for yesterday's pagination? Probably a bug

Reply:
Yep, huge bug, nice catch.